ext/updater.go

```diff
@@ -352,27 +353,41 @@ func (u *Updater) StartServer(opts WebhookOpts) error {
 	}
 
 	var tls bool
-	if opts.CertFile == "" && opts.KeyFile == "" {
+	switch {
+	case opts.CertFile == "" && opts.KeyFile == "":
 		tls = false
-	} else if opts.CertFile != "" && opts.KeyFile != "" {
+	case opts.CertFile != "" && opts.KeyFile != "":
 		tls = true
-	} else {
+	default:
 		return ErrMissingCertOrKeyFile
 	}
 
+	var network string
+	if opts.UnixListen != "" {
+		network = "unix"
+	} else {
+		network = "tcp"
+	}
+
```

Comment:
I think you're right, this is too convoluted. I was trying to be backward-compatible, but since we are still in rc, I think we should:

- remove the `Port` and `UnixListen` fields and add a `ListenNet` field.
- `ListenNet` can take "tcp", "tcp4", tcp6", "unix", "unixpacket" values. If empty, assume "tcp".
- the `Listen` field uses a "host:port" or a path syntax, depending on `ListenNet`.